src/cast_core/src/CrudActor.cpp

```diff
@@ -1164,14 +1247,20 @@ struct CrudActor::CollectionName {
 
 struct CrudActor::PhaseConfig {
     std::vector<std::unique_ptr<BaseOperation>> operations;
+    std::vector<std::unique_ptr<State>> states;
+    std::vector<double> initial_state_weights;
     metrics::Operation metrics;
     std::string dbName;
     CrudActor::CollectionName collectionName;
+    bool continue_current_state;
```

Comment:
Consider a `StateConfig` "sub-struct" that holds all the states/weights info together distinct from the `PhaseConfig`.